### ISSUE TITLE
ssh error when I add more hosts

### ISSUE THREAD

**Issue by villins** [capistrano/capistrano#1534]
Before I had 4 servers, but now I add more 10 servers.
I am trying to deploy all 14 servers, git check heads and pull code from my gitlab server,but I got some problem, as follows:
```
DEBUG [49ca967f] 	ssh_exchange_identification: Connection closed by remote host
DEBUG [49ca967f] 	fatal: Could not read from remote repository.
DEBUG [49ca967f]
DEBUG [49ca967f] 	Please make sure you have the correct access rights
DEBUG [49ca967f] 	and the repository exists.
```

The detail:
```
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing as deploy@server: git exit status: 128
git stdout: Nothing written
git stderr: ssh_exchange_identification: Connection closed by remote host
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/sshkit-1.7.1/lib/sshkit/runners/parallel.rb:16:in `rescue in block (2 levels) in execute'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/sshkit-1.7.1/lib/sshkit/runners/parallel.rb:12:in `block (2 levels) in execute'
SSHKit::Command::Failed: git exit status: 128
git stdout: Nothing written
git stderr: ssh_exchange_identification: Connection closed by remote host
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/sshkit-1.7.1/lib/sshkit/command.rb:95:in `exit_status='
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/sshkit-1.7.1/lib/sshkit/backends/netssh.rb:179:in `block in _execute'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/sshkit-1.7.1/lib/sshkit/backends/netssh.rb:133:in `tap'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/sshkit-1.7.1/lib/sshkit/backends/netssh.rb:133:in `_execute'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/sshkit-1.7.1/lib/sshkit/backends/netssh.rb:66:in `execute'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/capistrano-3.4.0/lib/capistrano/git.rb:11:in `git'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/capistrano-3.4.0/lib/capistrano/git.rb:21:in `check'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/capistrano-3.4.0/lib/capistrano/tasks/git.rake:28:in `block (4 levels) in <top (required)>'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/sshkit-1.7.1/lib/sshkit/backends/abstract.rb:85:in `with'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/capistrano-3.4.0/lib/capistrano/tasks/git.rake:27:in `block (3 levels) in <top (required)>'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/sshkit-1.7.1/lib/sshkit/backends/netssh.rb:54:in `instance_exec'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/sshkit-1.7.1/lib/sshkit/backends/netssh.rb:54:in `run'
/Users/marcolee/.rvm/gems/ruby-2.1.2/gems/sshkit-1.7.1/lib/sshkit/runners/parallel.rb:13:in `block (2 levels) in execute'
Tasks: TOP => git:check
The deploy has failed with an error: Exception while executing as deploy@server: git exit status: 128
git stdout: Nothing written
git stderr: ssh_exchange_identification: Connection closed by remote host
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
** Invoke deploy:failed (first_time)
** Execute deploy:failed
** Invoke bearychat:notify_deploy_fail (first_time)
** Execute bearychat:notify_deploy_fail
```
I try to separate deployment of the 4 old servers and the 10 new servers, it works.

What's the problem? And how to fix it?

Thx for rending my issue!

**Comment by mattbrictson** [comment]
Assuming you are running your own gitlab server, this might be because your gitlab can't handle the additional connections. The way Capistrano works, all 14 machines will try to check out from gitlab in parallel, which may be beyond the server's limits.

You may want check out the sequential git checkout workaround discussed here: https://github.com/capistrano/capistrano/issues/1058.

**Comment by mattbrictson** [comment]
I'm closing this issue due to inactivity and since it is probably a duplicate of #1058.
